src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -459,16 +459,16 @@ boolean containsLocal(
       accessed(foundDigests);
     }
     ImmutableList<Digest> missingDigests = builder.build();
-    if (delegate != null && !missingDigests.isEmpty()) {
-      return delegate.findMissingBlobs(missingDigests);
-    }
-    return missingDigests;
+    return CasFallbackHandler.findMissingBlobs(delegate, missingDigests);
   }
 
   @Override
   public boolean contains(Digest digest, Digest.Builder result) {
-    return containsLocal(digest, result, (key) -> accessed(ImmutableList.of(key)))
-        || (delegate != null && delegate.contains(digest, result));
+    if (containsLocal(digest, result, (key) -> accessed(ImmutableList.of(key)))) {
+      return true;
+    }
+
+    return CasFallbackHandler.contains(delegate, digest, result);
```

Comment:
feel free to do this, but leave the original logic. single return and all.

Reply:
fixed